Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py

```diff
@@ -238,45 +254,140 @@ def get_email_related_incident_id(email_related_incident_code, email_original_su
+                # If no related thread is found, determine the highest thread number
+                max_thread_number = 0
+                for message in incident_email_threads:
+                    if int(message['EmailCommsThreadNumber']) > max_thread_number:
+                        max_thread_number = int(message['EmailCommsThreadNumber'])
+
+                thread_number = str(max_thread_number + 1)
+        else:
+            thread_number = '0'
+
+        if len(thread_number) == 0:
+            demisto.error('Failed to identify a Thread Number to set. Email not appended to incident context')
+
+        if attachments:
+            attachment_names = [attachment.get('name', '') for attachment in attachments]
+        else:
+            attachment_names = ["None"]
+
+        email_message = {
+            'EmailCommsThreadId': email_code,
+            'EmailCommsThreadNumber': thread_number,
+            'EmailCC': email_cc,
+            'EmailBCC': email_bcc,
+            'EmailBody': email_text,
+            'EmailFrom': email_from,
+            'EmailHTML': email_html,
+            'MessageID': email_latest_message,
+            'EmailReceived': email_received,
+            'EmailReplyTo': email_replyto,
+            'EmailSubject': email_subject,
+            'EmailTo': email_to,
+            'EmailAttachments': f'{attachment_names}',
+            'MessageDirection': 'inbound',
+            'MessageTime': get_utc_now().strftime("%Y-%m-%dT%H:%M:%SUTC")
+        }
+        # Add email message to context key
+        try:
+            demisto.executeCommand('executeCommandAt', {
+                'command': 'Set', 'incidents': incident_id, 'arguments':
+                    {'key': 'EmailThreads', 'value': email_message, 'append': 'true'}})
+        except Exception as e:
+            demisto.error(f"Failed to append new email to context of incident {incident_id}. Reason: {e}")
+    except Exception as e:
+        demisto.error(f"Unable to add new email message to Incident {incident_id}. Reason: \n {e}")
+
+
 def main():
     incident = demisto.incident()
     custom_fields = incident.get('CustomFields')
     email_from = custom_fields.get('emailfrom')
     email_cc = custom_fields.get('emailcc')
+    email_bcc = custom_fields.get('emailbcc')
     email_to = custom_fields.get('emailto')
     email_subject = custom_fields.get('emailsubject')
+    email_text = custom_fields.get('emailbody')
     email_html = custom_fields.get('emailhtml')
+    email_received = custom_fields.get('emailreceived')
+    email_replyto = custom_fields.get('emailreplyto')
     attachments = incident.get('attachment', [])
     email_latest_message = custom_fields.get('emaillatestmessage')
 
     try:
         email_related_incident_code = email_subject.split('<')[1].split('>')[0]
-        email_original_subject = email_subject.split('<')[-1].split('>')[1]
+        email_original_subject = email_subject.split('<')[-1].split('>')[1].strip()
```

Comment:
Without the `strip()` the `email_original_subject` would start with a space, since the full subject will be in this format:
<12345678> Email Subject Text

With this change `email_original_subject` will not start with a space.